api/lib/domain/services/user-reconciliation-service.js

```diff
@@ -62,49 +60,36 @@ async function findMatchingSchoolingRegistrationIdForGivenOrganizationIdAndUser(
+  return _.find(schoolingRegistrations, { 'id': schoolingRegistrationId });
 }
 
-async function checkIfStudentIsAlreadyReconciledOnTheSameOrganization(matchingSchoolingRegistration, userRepository, obfuscationService) {
-  if (!_.isNil(matchingSchoolingRegistration.userId))  {
-    const userId = matchingSchoolingRegistration.userId ;
-    const user = await userRepository.getUserAuthenticationMethods(userId);
-    const authenticationMethod = obfuscationService.getUserAuthenticationMethodWithObfuscation(user);
-
-    const detail = 'Un compte existe déjà pour l‘élève dans le même établissement.';
-    const error = STUDENT_RECONCILIATION_ERRORS.RECONCILIATION.IN_SAME_ORGANIZATION[authenticationMethod.authenticatedBy];
-    const meta = {
-      shortCode: error.shortCode,
-      value: authenticationMethod.value,
-      userId: userId,
-    };
-    if (authenticationMethod.authenticatedBy === 'samlId') {
-      meta.schoolingRegistrationId = matchingSchoolingRegistration.id;
-    }
-    throw new SchoolingRegistrationAlreadyLinkedToUserError(detail, error.code, meta);
+async function checkIfStudentHasAnAlreadyReconciledAccount(schoolingRegistration, userRepository, obfuscationService, studentRepository) {
+  if (!_.isNil(schoolingRegistration.userId)) {
+    await _buildStudentReconciliationError(schoolingRegistration.userId, 'IN_SAME_ORGANIZATION', userRepository, obfuscationService);
   }
-}
 
-async function checkIfStudentHasAlreadyAccountsReconciledInOtherOrganizations(student, userRepository, obfuscationService) {
+  const student = await studentRepository.getReconciledStudentByNationalStudentId(schoolingRegistration.nationalStudentId);
   if (_.get(student, 'account')) {
-    const userId = student.account.userId;
-    const user = await userRepository.getUserAuthenticationMethods(userId);
-    const authenticationMethod = obfuscationService.getUserAuthenticationMethodWithObfuscation(user);
-
-    const detail = 'Un compte existe déjà pour l‘élève dans un autre établissement.';
-    const error = STUDENT_RECONCILIATION_ERRORS.RECONCILIATION.IN_OTHER_ORGANIZATION[authenticationMethod.authenticatedBy];
-    const meta = {
-      shortCode: error.shortCode,
-      value: authenticationMethod.value,
-      userId: userId,
-    };
-    throw new SchoolingRegistrationAlreadyLinkedToUserError(detail, error.code, meta);
+    await _buildStudentReconciliationError(student.account.userId, 'IN_OTHER_ORGANIZATION', userRepository, obfuscationService);
   }
 }
 
+async function _buildStudentReconciliationError(userId, errorContext, userRepository, obfuscationService) {
+  const user = await userRepository.getUserAuthenticationMethods(userId);
+  const authenticationMethod = obfuscationService.getUserAuthenticationMethodWithObfuscation(user);
+
+  const detailVariant = errorContext === 'IN_SAME_ORGANIZATION' ? 'le même' : 'un autre';
+  const detail = `Un compte existe déjà pour l‘élève dans ${detailVariant} établissement.`;
```

Comment:
C'est du détail mais j'ai un peu de mal à la lecture avec ce genre de pratique en général (conditionner une partie du contenu d'un texte). Perso je préfère un truc comme ça 

```suggestion
  const detailVariant = errorContext === 'IN_SAME_ORGANIZATION' ? 'le même' : 'un autre';
  const detail = `Un compte existe déjà pour l‘élève dans ${detailVariant} établissement.`;
```
```suggestion

  const detailWhenSameOrganization = `Un compte existe déjà pour l‘élève dans le même établissement.`;
  const detailWhenOtherOrganization = `Un compte existe déjà pour l‘élève dans un autre établissement.`;
  const detail = errorContext === 'IN_SAME_ORGANIZATION' ? detailWhenSameOrganization : detailWhenOtherOrganization;
```

Reply:
Done.